Assembly/Scripts/Characters/Human/Human.cs

```diff
@@ -274,9 +274,37 @@ public void SpecialActionState(float time)
         public void TransformShifter(string shifter, float liveTime)
         {
             _inGameManager.SpawnPlayerShifterAt(shifter, liveTime, Cache.Transform.position);
+            ((BaseShifter)_inGameManager.CurrentCharacter).PreviousHumanGas = CurrentGas;
+            ((BaseShifter)_inGameManager.CurrentCharacter).PreviousHumanWeapon = Weapon;
             PhotonNetwork.Destroy(gameObject);
         }
 
+        public IEnumerator WaitAndTransformFromShifter(float previousHumanGas, BaseUseable previousHumanWeapon)
+        {
+            while (!FinishSetup)
+            {
+                yield return null;
+            }
+            CurrentGas = previousHumanGas;
+            if (previousHumanWeapon is BladeWeapon)
+            {
+                ((BladeWeapon)Weapon).BladesLeft = ((BladeWeapon)previousHumanWeapon).BladesLeft;
+                ((BladeWeapon)Weapon).CurrentDurability = ((BladeWeapon)previousHumanWeapon).CurrentDurability;
+                if (((BladeWeapon)Weapon).CurrentDurability == 0)
```

Comment:
for bladeweapon and ammoweapon, if we're going to be casting 3 or more times lets declare a new variable so we only cast once. Casting is bad for performance if we do it too much